x-pack/plugins/infra/public/components/logging/log_analysis_setup/setup_flyout/module_list_card.tsx

```diff
@@ -26,19 +27,46 @@ export const LogAnalysisModuleListCard: React.FC<{
   moduleStatus,
   onViewSetup,
 }) => {
+  const {
+    services: {
+      ml,
+      application: { navigateToUrl },
+      notifications: { toasts },
+    },
+  } = useKibanaContextForPlugin();
+
+  const [viewInMlLink, setViewInMlLink] = useState<string>('');
+
+  const getMlUrl = async () => {
+    if (!ml.urlGenerator) {
+      toasts.addWarning({
+        title: mountReactNode(
+          <FormattedMessage
+            id="xpack.infra.logs.analysis.mlNotAvailable"
+            defaultMessage="ML plugin is not available"
+          />
+        ),
+      });
+      return;
+    }
+    setViewInMlLink(await ml.urlGenerator.createUrl({ page: 'jobs', pageState: { jobId } }));
```

Comment:
Unrelated to this PR but I'm curious: how come the `createUrl()` method is `async`? Does it fetch anything in the background to generate the URL?

Reply:
It's a question for the Kibana team. Check the exposed public contract of the URL generator https://github.com/elastic/kibana/blob/543481ba539b5003bbb8f7b2ed987a7c747c964d/src/plugins/share/public/url_generators/url_generator_internal.ts#L73